core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SignForIdentifierHandler.java

```diff
@@ -198,12 +192,31 @@ private boolean maySign(
     }
   }
 
+  private UInt64 getBlockSlot(final Eth2SigningRequestBody eth2SigningRequestBody) {
+    if (eth2SigningRequestBody.getBlock() != null) {
```

Comment:
There are a lot null checks here, can we use Optional instead?

Reply:
refactored to use specMilestone instead.